src/parsers/pm2ps_parser.jl

```diff
@@ -115,112 +52,176 @@ function make_bus(bus_name, bus_number, d, bus_types)
+
+        for (key, load) in data["load"]
+            load_bus = bus_number_to_bus[load["load_bus"]]
+            if get_name(load_bus) in bus_names
+                push!(active_power, load["pd"])
+                push!(reactive_power, load["qd"])
+            end
+        end
+
+        load_zones = make_loadzones(d_key, d, buses, active_power, reactive_power)
+        add_component!(sys, load_zones)
     end
 end
 
 function make_hydro_gen(gen_name, d, bus)
     ramp_agc = get(d, "ramp_agc", get(d, "ramp_10", get(d, "ramp_30", d["pmax"])))
-    hydro =  Dict{String,Any}("name" => gen_name,
-                            "available" => d["gen_status"], # change from staus to available
-                            "bus" => bus,
-                            "tech" => Dict{String,Any}( "rating" => sqrt(d["pmax"]^2 + d["qmax"]^2),
-                                                        "activepower" => d["pg"],
-                                                        "activepowerlimits" => (min=d["pmin"], max=d["pmax"]),
-                                                        "reactivepower" => d["qg"],
-                                                        "reactivepowerlimits" => (min=d["qmin"], max=d["qmax"]),
-                                                        "ramplimits" => (up=ramp_agc/d["mbase"],down=ramp_agc/d["mbase"]),
-                                                        "timelimits" => nothing),
-                            "econ" => Dict{String,Any}("curtailcost" => 0.0,
-                                                        "interruptioncost" => nothing)
-                            )
-    return hydro
+    tech = TechHydro(;
+        rating=calculate_rating(d["pmax"], d["qmax"]),
+        activepower=d["pg"],
+        activepowerlimits=(min=d["pmin"], max=d["pmax"]),
+        reactivepower=d["qg"],
+        reactivepowerlimits=(min=d["qmin"], max=d["qmax"]),
+        ramplimits=(up=ramp_agc / d["mbase"], down=ramp_agc / d["mbase"]),
+        timelimits=nothing,
+    )
+
+    curtailcost = 0.0
+
+    return HydroDispatch(gen_name, Bool(d["gen_status"]), bus, tech, curtailcost)
 end
 
-function make_ren_gen(gen_name, d, bus)
-    gen_re = Dict{String,Any}("name" => gen_name,
-                    "available" => d["gen_status"], # change from staus to available
-                    "bus" => bus,
-                    "tech" => Dict{String,Any}("rating" => float(d["pmax"]),
-                                                "reactivepowerlimits" => (min=d["pmin"], max=d["pmax"]),
-                                                "powerfactor" => 1),
-                    "econ" => Dict{String,Any}("curtailcost" => 0.0,
-                                                "interruptioncost" => nothing)
-                    )
-    return gen_re
+function make_tech_renewable(d)
+    tech = TechRenewable(;
+        rating=float(d["pmax"]),
+        reactivepowerlimits=(min=d["qmin"], max=d["qmax"]),
```

Comment:
Something I'll call out is that the old dict_to_struct code ignored reactivepowerlimits, so even though there was matpower data, the TechRenewable got a nothing.  Now it will have something.  The matpower file case5_re.m has (0.0, 0.0).  The recent CDM parsing changes caused the same result.